src/NuGet.Services.BasicSearch/OwinRequestIdTelemetryInitializer.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace NuGet.Services.BasicSearch
+{
+    public class OwinRequestIdTelemetryInitializer
+        : ITelemetryInitializer
+    {
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (OwinRequestIdContext.Get() != null)
```

Comment:
Maybe store the result of `OwinRequestIdContext.Get()` in a local field, instead of calling it twice?